lib/if.c

```diff
@@ -1141,7 +1141,11 @@ DEFPY_NOSH (interface,
 	 * VRF.
 	 */
 	VRF_GET_ID(vrf_id, vrfname, false);
-	ifp = if_lookup_by_name_all_vrf(ifname);
+	if (vrf_get_backend() == VRF_BACKEND_UNKNOWN ||
```

Comment:
As I have stated elsewhere, it is not appropriate for end programmer to know *which* interface lookup function they must use when.  Can we get this right please?

Reply:
For that, I made a new static routine that will wrap the modification in if_lookup_by_name_relax(). This is mainly to avoid breaking behaviour.